### PR TITLE
Attempt to catch user vs org repos

### LINKED ISSUE #62: using `orgs` API to try to access user repositories
The website builds are currently failing, due to the changes made to `feeds/other_lessons.R` in [13a0ea3](https://github.com/hpc-carpentry/hpc-carpentry.github.io/commit/13a0ea3a290c17a5b881f00283dca2adef8822e3).

The problem is that the `get_org_topics` function used on line 47 of that file

```
  frobnitzem <- get_org_topics("frobnitzem")
```

uses [the `get_list_repos` function defined in `feeds/utils.R`](https://github.com/hpc-carpentry/hpc-carpentry.github.io/blob/7002f34d8556b804816fe91457cf09d9f942a1d0/feeds/utils.R#L22), which makes a call to the GitHub _organizations_ API, to try to get a list of repositories for an organisation. `frobnitzem` is a _user_ account, so should be queried via the _users_ API. See the difference between the current call that is being made

https://api.github.com/orgs/frobnitzem/repos

and the result of the equivalent call made via the users API

https://api.github.com/users/frobnitzem/repos

I am fairly certain that this is the _problem_, but less confident about the best _solution_. I suppose you would need to adjust `get_list_repos` to accept an optional parameter that tells it to query the users API instead of the orgs API?

### PR DESCRIPTION
**Pull request by tkphd**
Closes #62.

### CODE REVIEW

**Comment by tkphd** [comment]
@ocaisa please take a look

**Review by ocaisa — APPROVED** [review]
I _think_ I understand this but I'm really not very familiar with R. The proof of the pudding is in eating so going to merge this.

File: feeds/utils.R [hunk 1]
```diff
@@ -19,10 +19,14 @@ use_pat <- function() {
...
-
-  init_res  <- gh::gh("GET /orgs/:org/repos", org = org)
+get_list_repos <- function(org,
+                           org_is_user = FALSE,
+                           ignore_archived = FALSE,
+                           ignore_pattern = NULL,
+                           ...) {
+
+  init_res  <- if(org_is_user) gh::gh("GET /users/:org/repos", org = org)
+               else gh::gh("GET /orgs/:org/repos", org = org)
```
Comment (ocaisa) [inline feeds/utils.R:29]: @tkphd Unfortunately, this is failing, see https://github.com/hpc-carpentry/hpc-carpentry.github.io/actions/runs/3386633494/jobs/5626289198#step:9:47. I think these are the wrong way around?

Comment (ocaisa) [reply feeds/utils.R:29]: Fixed in #65

File: feeds/utils.R [hunk 2]
```diff
@@ -110,6 +115,20 @@ get_org_topics <- function(org) {
     )
 }
 
+get_usr_topics <- function(usr) {
+  # User should be lower case
+  usr <- tolower(usr)
+  get_list_repos(usr) %>%
+    dplyr::filter(
+      !private,
+      carpentries_usr == usr
```
Comment (ocaisa) [inline feeds/utils.R:124]: @tkphd I get the feeling here that the filter here should remain `carpentries_org` (which is useless in this case)

Comment (ocaisa) [reply feeds/utils.R:124]: Fixed in https://github.com/hpc-carpentry/hpc-carpentry.github.io/commit/87cda90ead2d4dbee83c5f80927b050344d5f8cd

**Pull request merged**